enterprise/wsproxy/wsproxy_test.go

```diff
@@ -442,6 +442,13 @@ func TestWorkspaceProxyWorkspaceApps(t *testing.T) {
 			"*",
 		}
 
+		proxyFlushCh := make(chan chan<- struct{})
+		flushStats := func() {
+			proxyFlushDone := make(chan struct{})
+			proxyFlushCh <- proxyFlushDone
+			<-proxyFlushDone
+		}
```

Comment:
review: there should be no need to flush the stats on the primary here, as the proxy sends its stats directly to the primary endpoint which is wired up directly to the Reporter, which just inserts immediately.

Reply:
Correct me if I'm wrong, but without a flush, we can't ensure that `Report()` is being called (i.e. flushing the in-memory stats to coderd via post), right? So I think it may be needed here too, depending on what we're looking to achieve.

Reply:
From my reading, any POSTS to `/app-stats` go straight to `api.WorkspaceAppsStatsCollectorOptions.Reporter`, and the wsproxy app stats reporter just hits that endpoint on flush.